### PR TITLE
delete branches after merge, enable automatic merging of prs

### PR DESCRIPTION
**Pull request by ghuntley**
## PR Type
What kind of change does this PR introduce?

- Build or CI related changes

## What is the current behavior?

- Maintainers need to babysit builds and wait for them to pass before merging.
- Maintainers need to remember to delete branches after they are merged.

## What is the new behavior?

- When a pull-request to `master` has 2 or more approvals and no requests for change it will be automatically merged if the CI passes, is labelled with `ready-to-merge` unless it has been labelled with `do-not-merge/breaking-changes` or `do-not-merge/work-in-progress`

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

See https://github.com/digital-asset/daml/blob/master/.mergify.yml for comparison.
See https://doc.mergify.io/configuration.html for documentation

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      # Note that this only matches people with write / admin access to the repo,
+      # see <https://doc.mergify.io/conditions.html#attribute-list>
+      - "#approved-reviews-by>=2"
+
+      # Note: mergify cannot break branch protection rules
+    actions:
+      merge:
+        method: merge
+
+#  - name: automatic removal of possibly outdated reviews approvals or changes request when a pull request gets updated with new commits
```
Comment (ghuntley) [inline .mergify.yml]: I think, for now, we should keep this option disabled as it provides the ability for folks to squash their commits after approvals have been done.

**Comment by MatFillion** [comment]
Is it possible to automatically merge only if the author has set to auto-complete ? (a bit like Azure Devops). I'm a bit afraid that a PR I would like to squash some commits after reviews would automatically get merged before I get the chance to do so.

**Comment by davidjohnoliver** [comment]
> Is it possible to automatically merge only if the author has set to auto-complete ? (a bit like Azure Devops). I'm a bit afraid that a PR I would like to squash some commits after reviews would automatically get merged before I get the chance to do so.
> 
> Is is possible to require all conversations to be resolved as well ?

Yeah having the conversations resolved is a must

**Comment by carldebilly** [comment]
@MatFillion: I think it would make sense to use draft PR intensively when not ready to be merged.

**Comment by ghuntley** [comment]
# proposal that incorporates feedback from above

Configure MergeBot to NOT automatically merge unless:

- approved 2 or more reviewers
- CI passes
- require all require all request for changes to be resolved
- has label the `automerge` applied

# additional configuration options

## strict merging

By enabling strict merging with the `smart` mode mergabe pull-requests will be updated automatically with the contents of master. If CI passes, then Mergify will merge that pull-request. If multiple pull-requests are mergable open then Mergify will queue the mergeable pull requests and update them one at a time serially.

This prevents merging broken pull requests:

https://doc.mergify.io/strict-workflow.html#strict-merge

For more information about merge behaviour configuration see:

https://doc.mergify.io/actions.html#merge

🙋 _I think we should enable this option. Thoughts?_

## automatically removing stale reviews

When a pull request is updated, GitHub does not remove the possibly outdated reviews approvals or changes request. This behavior can be changed to dismiss all reviews when new commits are pushed.

https://doc.mergify.io/examples.html#id7

🙋 _Whilst this introduces more safety, it could be annoying. Thoughts?_

**Comment by MatFillion** [comment]
I like it :)

For your question, I would not reset approvals on  branch changes.

**Comment by MatFillion** [comment]
For the strict merge question, yes we should re-run CI builds whenever master is updated(and prevent merges if the ci build isn't valid anymore). But it is already the case in the current setup (without mergify). Unless I misunderstood your question?

**Comment by ghuntley** [comment]
> It is already the case in the current setup (without mergify).

Doesn't look like it is. I've been able to merge PR's which where behind head /cc @carldebilly

**Comment by ghuntley** [comment]
Feedback has been incorporated and the configuration has been updated. Merging this pull-request will activate mergify 🎉

**Comment by ghuntley** [comment]
Feedback has been incorporated and the configuration has been updated. Merging this pull-request will activate mergify 🎉

**Comment by davidjohnoliver** [comment]
I think we should add this condition: `"#changes-requested-reviews-by=0"` https://doc.mergify.io/conditions.html#examples

**Comment by ghuntley** [comment]
> I think we should add this condition: "#changes-requested-reviews-by=0"

Yup. Adjusted.

**Pull request merged**